deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/layers/Convolution1DLayer.java

```diff
@@ -0,0 +1,156 @@
+        super(builder);
+    }
+
+    @Override
+    public org.deeplearning4j.nn.api.Layer instantiate(NeuralNetConfiguration conf, Collection<IterationListener> iterationListeners, int layerIndex, INDArray layerParamsView, boolean initializeParams) {
+        org.deeplearning4j.util.LayerValidation.assertNInNOutSet("Convolution1DLayer", getLayerName(), layerIndex, getNIn(), getNOut());
+
+        org.deeplearning4j.nn.layers.convolution.Convolution1DLayer ret
+                = new org.deeplearning4j.nn.layers.convolution.Convolution1DLayer(conf);
+        ret.setListeners(iterationListeners);
+        ret.setIndex(layerIndex);
+        ret.setParamsViewArray(layerParamsView);
+        Map<String, INDArray> paramTable = initializer().init(conf, layerParamsView, initializeParams);
+        ret.setParamTable(paramTable);
+        ret.setConf(conf);
+        return ret;
+    }
+
+    @Override
+    public InputType getOutputType(int layerIndex, InputType inputType) {
+        if (inputType == null || inputType.getType() != InputType.Type.RNN) {
+            throw new IllegalStateException("Invalid input for 1D CNN layer (layer index = " + layerIndex +
+                    ", layer name = \"" + getLayerName() + "\"): expect RNN input type with size > 0. Got: " + inputType);
+        }
+
+        return InputType.recurrent(nOut);
+    }
+
+    @Override
+    public void setNIn(InputType inputType, boolean override){
+        if (inputType == null || inputType.getType() != InputType.Type.RNN) {
+            throw new IllegalStateException("Invalid input for 1D CNN layer (layer name = \"" + getLayerName() + "\"): expect RNN input type with size > 0. Got: " + inputType);
+        }
+
+        if (nIn <= 0 || override) {
+            InputType.InputTypeRecurrent r = (InputType.InputTypeRecurrent) inputType;
+            this.nIn = r.getSize();
+        }
+    }
+
+    @Override
+    public InputPreProcessor getPreProcessorForInputType(InputType inputType) {
+        if(inputType == null ){
+            throw new IllegalStateException("Invalid input for Convolution1D layer (layer name=\"" + getLayerName() + "\"): input is null");
+        }
+
+        return InputTypeUtil.getPreprocessorForInputTypeRnnLayers(inputType, getLayerName());
+    }
+
+    public static class Builder extends ConvolutionLayer.Builder {
+        public Builder(int kernelSize, int stride, int padding) {
+            this();
+            this.kernelSize[0] = kernelSize;
+            this.stride[0] = stride;
+            this.padding[0] = padding;
+        }
+
+        public Builder(int kernelSize, int stride) {
+            this();
+            this.kernelSize[0] = kernelSize;
+            this.stride[0] = stride;
+        }
+
+        public Builder(int kernelSize) {
+            this();
+            this.kernelSize[0] = kernelSize;
+        }
+
+        public Builder() {
+            this.kernelSize = new int[] { 5, 1 };
+            this.stride = new int[] { 1, 1 };
```

Comment:
Fixed. Now initialized to `null`. When passed to constructor, an Exception is throw. Added a unit test to validate this.